From: Remi Collet <fedora@famillecollet.com>
Date: Thu, 30 Mar 2017 09:21:27 +0200
Subject: preserve BC after fix for #74216

x
---
 ext/standard/http_fopen_wrapper.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/ext/standard/http_fopen_wrapper.c b/ext/standard/http_fopen_wrapper.c
index ed6adc0..3ad5f68 100644
--- a/ext/standard/http_fopen_wrapper.c
+++ b/ext/standard/http_fopen_wrapper.c
@@ -190,8 +190,13 @@ php_stream *php_stream_url_wrap_http_ex(php_stream_wrapper *wrapper,
 			Z_TYPE_PP(tmpzval) == IS_STRING &&
 			Z_STRLEN_PP(tmpzval) > 0) {
 			use_proxy = 1;
-			transport_len = Z_STRLEN_PP(tmpzval);
-			transport_string = estrndup(Z_STRVAL_PP(tmpzval), Z_STRLEN_PP(tmpzval));
+			if (Z_STRLEN_P(tmpzval) > 2 && Z_STRVAL_P(tmpzval)[Z_STRLEN_P(tmpzval)-1] == '/') {
+				/* ignore trailing slash to preserve BC */
+				transport_len = Z_STRLEN_P(tmpzval)-1;
+			} else {
+				transport_len = Z_STRLEN_P(tmpzval);
+			}
+			transport_string = estrndup(Z_STRVAL_P(tmpzval), transport_len);
 		} else {
 			transport_len = spprintf(&transport_string, 0, "%s://%s:%d", use_ssl ? "ssl" : "tcp", resource->host, resource->port);
 		}
